src/commands/external.rs

```diff
@@ -0,0 +1,75 @@
+use anyhow::{anyhow, Result};
+use clap::App;
+use spin_plugins::{manifest::check_supported_version, PluginStore};
+use std::{collections::HashMap, env, process};
+use tokio::process::Command;
+use tracing::log;
+
+/// Executes a Spin plugin as a subprocess, expecting the first argument to
+/// indicate the plugin to execute. Passes all subsequent arguments on to the
+/// subprocess.
+pub async fn execute_external_subcommand(args: Vec<String>, app: App<'_>) -> Result<()> {
+    let plugin_name = args.first().ok_or_else(|| anyhow!("Expected subcommand"))?;
```

Comment:
clap doesn't allow defining complex types for `external_subcommand`: `The type must be either `Vec<String>` or `Vec<OsString>` to be used with `external_subcommand`.` It would be preferable to have the cli input immediately parsed by clap as a command and vector of args but clap doesnt support this.
I did try to make this cleaner by clearly separating the command/plugin name from the rest of the arguments in the first line:
```
let (plugin_name, args) = cmd
        .split_first()
        .ok_or_else(|| anyhow!("Expected subcommand"))?;
```